satellite/audit/service.go

```diff
@@ -63,13 +79,36 @@ func (service *Service) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	service.log.Info("Audit cron is starting up")
 
-	return service.Loop.Run(ctx, func(ctx context.Context) error {
-		err := service.process(ctx)
-		if err != nil {
-			service.log.Error("process", zap.Error(err))
-		}
-		return nil
+	group, ctx := errgroup.WithContext(ctx)
+
+	group.Go(func() error {
+		return service.Loop.Run(ctx, func(ctx context.Context) (err error) {
+			defer mon.Task()(&ctx)(&err)
+			err = service.process(ctx)
+			if err != nil {
+				service.log.Error("process", zap.Error(err))
+			}
+			return nil
+		})
 	})
+
+	group.Go(func() error {
+		return service.ReservoirLoop.Run(ctx, func(ctx context.Context) (err error) {
+			defer mon.Task()(&ctx)(&err)
+			observer := NewObserver(service.log.Named("audit observer"), service.overlay, service.reservoirSlots)
+			err = service.MetainfoLoop.Join(ctx, observer)
+			if err != nil {
+				service.log.Error("error joining metainfoloop", zap.Error(err))
+				return nil
+			}
+			for nodeID, res := range observer.Reservoirs {
```

Comment:
I think you should just be able to say `service.Reservoirs = observer.Reservoirs` instead of iterating over every item

Reply:
note for other people thinking this- the reason I looped through is because `Reservoirs` is a map, which is a reference type, so iterating through is necessary to properly copy the values and not just create a reference to the observer.Reservoirs map.